cpp/src/stream_compaction/stream_compaction.cu

```diff
@@ -108,4 +114,136 @@ gdf_column drop_nulls(gdf_column const &input) {
   }
 }
 
+
+namespace detail {
+
+/*
+ * unique_copy copies elements from the range [first, last) to a range beginning
+ * with output, except that in a consecutive group of duplicate elements only
+ * depending on last argument keep, only the first one is copied, or the last
+ * one is copied or neither is copied. The return value is the end of the range
+ * to which the elements are copied.
+ */
+template<typename DerivedPolicy,
+         typename InputIterator,
+         typename OutputIterator,
+         typename BinaryPredicate,
+    typename IndexType = typename
+  thrust::iterator_difference<InputIterator>::type>
+  OutputIterator unique_copy(thrust::execution_policy<DerivedPolicy> &exec,
+                             InputIterator first,
+                             InputIterator last,
+                             OutputIterator output,
+                             BinaryPredicate comp,
+                             const duplicate_keep_option keep)
+{
+  IndexType n = (last-first)-1;
+  if (keep == duplicate_keep_option::KEEP_FIRST) {
+      return thrust::copy_if(exec,
```

Comment:
If this used `cudf::copy_if` instead, it would automatically handle the output bitmasks for you.

Reply:
Disregard. I realized that there is no null bitmasks here since you're just doing a `copy_if` on indices.